internal/infrastructure/kubernetes/infra_client.go

```diff
@@ -58,3 +59,11 @@ func (cli *InfraClient) Delete(ctx context.Context, object client.Object) error
 
 	return nil
 }
+
+// GetUID retrieves the uid of one resource.
+func (cli *InfraClient) GetUID(ctx context.Context, key client.ObjectKey, current client.Object) (types.UID, error) {
```

Comment:
maybe instead of implement `GetUID` as the method of `InfraClient`, we can wrap it as a function, just like this function did:
https://github.com/envoyproxy/gateway/blob/7c6aa9ea849667d40d0c74c2acf37c8664068329/internal/provider/kubernetes/predicates.go#L301